components/automate-cli/cmd/chef-automate/summary.go

```diff
@@ -0,0 +1,429 @@
+package main
+
+import (
+	"container/list"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/chef/automate/components/automate-cli/pkg/status"
+	"github.com/chef/automate/lib/stringutils"
+	"github.com/jedib0t/go-pretty/v5/table"
+	"github.com/spf13/cobra"
+)
+
+var (
+	habUrl                = "https://localhost:9631"
+	clusterOSHealth       = "http://localhost:10144/_cluster/health"
+	a2haHabitatAutoTfvars = terraformPath + "/a2ha_habitat.auto.tfvars"
+	nowFunc               = time.Now
+)
+
+type StatusSummary interface {
+	Run(sshUtil SSHUtil) error
+	ShowFEStatus() string
+	ShowBEStatus() string
+}
+
+type Summary struct {
+	feStatus              FeStatus
+	beStatus              BeStatus
+	timeout               int64
+	spinnerTimeout        time.Duration
+	infra                 *AutomteHAInfraDetails
+	statusSummaryCmdFlags *StatusSummaryCmdFlags
+}
+
+type A2haHabitatAutoTfvars struct {
+	HabSupHttpGatewayAuthToken string `json:"hab_sup_http_gateway_auth_token"`
+	HabSupRingKey              string `json:"hab_sup_ring_key"`
+}
+
+// Display Frontend Status
+func (ss *Summary) ShowFEStatus() string {
+	if len(ss.feStatus) != 0 {
+		fmt.Println("Frontend Services")
+		t := table.NewWriter()
+		tTemp := table.Table{}
+		tTemp.Render()
+		for _, status := range ss.feStatus {
+			t.AppendRow(table.Row{status.serviceName, status.ipAddress, status.status, status.Opensearch})
+		}
+		t.AppendHeader(table.Row{"Name", "IP Address", "Status", "Opensearch"})
+		return t.Render()
+	}
+	return ""
+}
+
+// Display Backend Status
+func (ss *Summary) ShowBEStatus() string {
+	if len(ss.beStatus) != 0 {
+		fmt.Println("Backend Services")
+		t := table.NewWriter()
+		tTemp := table.Table{}
+		tTemp.Render()
+		for _, status := range ss.beStatus {
+			t.AppendRow(table.Row{status.serviceName, status.ipAddress, status.health, status.process, status.upTime, status.role})
+		}
+		t.AppendHeader(table.Row{"Name", "IP Address", "Health", "Process", "Uptime", "Role"})
+		return t.Render()
+	}
+	return ""
+}
+
+// Run Status summary
+func (ss *Summary) Run(sshUtil SSHUtil) error {
```

Comment:
`prepare` looks more appropriate name for this func instead of run, as it is just preparing some of scripts